opencti-platform/opencti-graphql/src/domain/user.js

```diff
@@ -153,9 +153,14 @@ const getUserAndGlobalMarkings = async (context, userId, userGroups, capabilitie
   return { user: computedMarkings, all: markings };
 };
 
-export const getCapabilities = async (context, userId, isUserPlatform) => {
-  const roles = await listThroughGetTo(context, SYSTEM_USER, userId, RELATION_HAS_ROLE, ENTITY_TYPE_ROLE);
-  const roleIds = roles.map((r) => r.id);
+export const getRoles = async (context, userGroups) => {
+  const groupIds = userGroups.map((r) => r.id);
+  const roles = await listThroughGetTo(context, SYSTEM_USER, groupIds, RELATION_HAS_ROLE, ENTITY_TYPE_ROLE);
+  return roles;
+};
+
+export const getCapabilities = async (context, userId, userRoles, isUserPlatform) => {
+  const roleIds = userRoles.map((r) => r.id);
   const capabilities = await listThroughGetTo(context, SYSTEM_USER, roleIds, RELATION_HAS_CAPABILITY, ENTITY_TYPE_CAPABILITY);
   if (userId === OPENCTI_ADMIN_UUID && !R.find(R.propEq('name', BYPASS))(capabilities)) {
```

Comment:
Maybe a uniq is needed on get capabilities as now duplicate can happen if user is part of groups that of roles with same capabilities

Reply:
fixed